Meter.c

```diff
@@ -268,90 +314,579 @@ static void BarMeterMode_draw(Meter* this, int x, int y, int w) {
 
 /* ---------- GraphMeterMode ---------- */
 
-#ifdef HAVE_LIBNCURSESW
+static void GraphMeterMode_createGraphData(Meter* this) {
+   assert(this->drawData == NULL);
+   uint8_t maxItems = Meter_maxItems(this);
+
+   // recordNumCells determines number of cells needed to allocate for a graph
+   // data record. A record corresponds to a terminal column when drawn.
+   size_t recordNumCells;
+   if (GRAPH_HEIGHT >= 2 && maxItems <= 1) {
+      // 1 item only
+      // Index 0: log2 of scale
+      // Index 1: Number of dots total
+      recordNumCells = 2;
+   } else if (this->total > 0.0) {
+      // Percent graph
+      // Index 0..GRAPH_HEIGHT: Color cells
+      recordNumCells = GRAPH_HEIGHT;
+   } else {
+      // Non-percentage & dynamic scale
+      // Index 0: log2 of scale
+      // Index 1..(2*GRAPH_HEIGHT-1): Color cells
+      recordNumCells = 2 * GRAPH_HEIGHT;
+   }
 
-#define PIXPERROW_UTF8 4
-static const char* const GraphMeterMode_dotsUtf8[] = {
-   /*00*/" ", /*01*/"⢀", /*02*/"⢠", /*03*/"⢰", /*04*/ "⢸",
-   /*10*/"⡀", /*11*/"⣀", /*12*/"⣠", /*13*/"⣰", /*14*/ "⣸",
-   /*20*/"⡄", /*21*/"⣄", /*22*/"⣤", /*23*/"⣴", /*24*/ "⣼",
-   /*30*/"⡆", /*31*/"⣆", /*32*/"⣦", /*33*/"⣶", /*34*/ "⣾",
-   /*40*/"⡇", /*41*/"⣇", /*42*/"⣧", /*43*/"⣷", /*44*/ "⣿"
-};
+   GraphData* data = xCalloc(1, sizeof(GraphData) +
+                                (METER_GRAPHDATA_SIZE * recordNumCells *
+                                   sizeof(GraphColorCell)));
```

Comment:
Please do two separate allocations for this and reference the second one from the main `GraphData` structure.
Even if this introduces the need for proper `dtor`s …

Reply:
Okay. Will change this in next commit.